rep-0144.rst

```diff
@@ -0,0 +1,129 @@
+REP: 144
+Title: ROS Package Naming and Versioning
+Author: Vincent Rabaud <vincent.rabaud@gmail.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 28-Jan-2015
+Post-History:
+
+
+Abstract
+========
+
+This REP gives advice on how a ROS package should be named and versioned. It
+formalizes and extends conventions that were formerly described in [1]_.
+
+Motivation
+==========
+
+As the number of ROS packages increases, it is hard to quickly find a package
+and guess its functionality based on its name. Over time, the lack of naming 
+conventions created problems like use of unexplained acronyms or
+packages with the same functionality but different names.
+
+This REP proposes rules to name a ROS package properly. Those 
+rules are not mandatory, merely informative.
+
+
+Package Naming
+==============
+
+Global Rules
+------------
+
+* ``alphanumerics`` are ``A-Za-z0-9`` only
+* a package name is defined more strictly than with Debian at [2]_. It must only consist
+  of lower case letters ``a-z``, digits ``0-9`` and ``_`` separators. They must be at
+  least two characters long and must start with an alphanumeric character
```

Comment:
According to the [Wiki naming conventions](http://wiki.ros.org/ROS/Patterns/Conventions#Packages),  package names *must* start with an alphabetic character, not a digit or underscore.

At one time there were a few package names starting with a digit, but don't think any of them are currently being released.